### PR TITLE
Remove origin shared informer factory and use build generated informers

### PR DESCRIPTION
**Pull request by smarterclayton**
Refactor builds to use generated informer

Still have to fix the tests

@mfojtik @csrwng @deads2k

### CODE REVIEW

File: pkg/build/controller/policy/policy.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ import (
 
 	buildapi "github.com/openshift/origin/pkg/build/api"
 	buildclient "github.com/openshift/origin/pkg/build/client"
+	buildlister "github.com/openshift/origin/pkg/build/generated/listers/build/internalversion"
```
Comment (csrwng) [inline pkg/build/controller/policy/policy.go]: @smarterclayton we should not move to using caches for the build policy stuff without considering all the implications. Here we really want to make sure that there's not another build in Running state before running another build (in the case of the serial policy). 

/cc @bparees

File: pkg/build/util/util.go [hunk 2]
```diff
@@ -114,25 +113,23 @@ func BuildConfigSelectorDeprecated(name string) labels.Selector {
...
 
 // BuildConfigBuilds return a list of builds for the given build config.
 // Optionally you can specify a filter function to select only builds that
 // matches your criteria.
-func BuildConfigBuilds(c buildclient.BuildLister, namespace, name string, filterFunc buildFilter) (*buildapi.BuildList, error) {
-	result, err := c.List(namespace, metav1.ListOptions{
-		LabelSelector: BuildConfigSelector(name).String(),
-	})
+func BuildConfigBuilds(c buildlister.BuildLister, namespace, name string, filterFunc buildFilter) ([]*buildapi.Build, error) {
+	result, err := c.Builds(namespace).List(BuildConfigSelector(name))
```
Comment (csrwng) [inline pkg/build/util/util.go]: should use an index?

File: pkg/build/controller/policy/policy.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/build/controller/policy/policy.go]: This is not moving to caches.  This is replacing the code that was using the dead hand written client.

File: pkg/build/util/util.go (hunk 2 above)
Comment (smarterclayton) [inline pkg/build/util/util.go]: Indexes are handled below this level

**Comment by smarterclayton** [comment]
[test]

File: pkg/build/controller/policy/policy.go (hunk 1 above)
Comment (csrwng) [inline pkg/build/controller/policy/policy.go]: The run policies then need to be instantiated with a buildLister created from a client and not the informer then:
https://github.com/openshift/origin/pull/14679/files#diff-4349d2367514c13a1d5c76dc4debf58fR126

File: pkg/build/util/util.go (hunk 2 above)
Comment (csrwng) [inline pkg/build/util/util.go]: Yeah, I meant should we introduce a new index to index builds by buildconfig key?

**Comment by smarterclayton** [comment]
Stomped a cache mutation

File: pkg/build/util/util.go (hunk 2 above)
Comment (smarterclayton) [inline pkg/build/util/util.go]: Namespace is probably sufficient - two indices might actually be slower.

File: pkg/build/controller/policy/policy.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/build/controller/policy/policy.go]: They aren't?  You can't create from an informer.

Comment (csrwng) [inline pkg/build/controller/policy/policy.go]: Maybe I'm saying the wrong thing ... but you're creating a buildLister in line 102 in build_controller.go. As far as I can tell, that's using a cache. 
```
+	buildLister := params.BuildInformer.Lister()
```

that's fine to use for build pruning, which we do use here https://github.com/smarterclayton/origin/blob/76f000014cf26e022b3b05c00bfe8b082429e291/pkg/build/controller/build/build_controller.go#L617

I think you need to create a separate one like:
```
policyBuildLister := buildclient.NewOSClientBuildLister(params.OpenshiftClient)
```
to pass to `policy.GetAllRunPolicies(...)`

Comment (smarterclayton) [inline pkg/build/controller/policy/policy.go]: Why would I need a separate one?

Comment (smarterclayton) [inline pkg/build/controller/policy/policy.go]: Oh, you're saying to do a live check.  Got it.

**Comment by smarterclayton** [comment]
flake in TestEtcd2StoragePath https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/2346/testReport/junit/github/com_openshift_origin_test_integration%20%20%20/TestEtcd2StoragePath/ @enj

[merge][severity:bug]

**Pull request merged**